http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcProviderMetadataTest.java

```diff
@@ -0,0 +1,374 @@
+    }
+
+    @Test
+    public void testPushedAuthorizationRequestEndpoint() throws Exception {
+        assertTrue("http://localhost:8080/realms/myrealm/protocol/openid-connect/ext/par/request".equals(oidcProviderMetadata.getPushedAuthorizationRequestEndpoint()));
+        assertNull(emptyOidcProviderMetadata.getPushedAuthorizationRequestEndpoint());
+    }
+
+    @Test
+    public void testRevocationEndpoint() throws Exception {
+        assertTrue("http://localhost:8080/realms/myrealm/protocol/openid-connect/revoke".equals(oidcProviderMetadata.getRevocationEndpoint()));
+        assertTrue("http://localhost:8080/revoke".equals(withoutBooleansOidcProviderMetadata.getRevocationEndpoint()));
+        assertNull(emptyOidcProviderMetadata.getRevocationEndpoint());
+    }
+
+    @Test
+    public void testRevocationEndpointAuthMethodsSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getRevocationEndpointAuthMethodsSupported();
+        assertTrue(l.contains("client_secret_basic"));
+        assertNull(emptyOidcProviderMetadata.getRevocationEndpointAuthMethodsSupported());
+    }
+
+    @Test
+    public void testRevocationEndpointAuthSigningAlgValuesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getRevocationEndpointAuthSigningAlgValuesSupported();
+        assertTrue(l.contains("RS384"));
+        assertNull(emptyOidcProviderMetadata.getRevocationEndpointAuthSigningAlgValuesSupported());
+    }
+
+    @Test
+    public void testBackchannelLogoutSupported() throws Exception {
+        assertTrue(oidcProviderMetadata.getBackchannelLogoutSupported());
+        assertFalse(withoutBooleansOidcProviderMetadata.getBackchannelLogoutSupported());
+    }
+
+    @Test
+    public void testBackchannelLogoutSessionSupported() throws Exception {
+        assertTrue(oidcProviderMetadata.getBackchannelLogoutSessionSupported());
+        assertFalse(withoutBooleansOidcProviderMetadata.getBackchannelLogoutSessionSupported());
+    }
+
+    @Test
+    public void testCodeChallengeMethodsSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getCodeChallengeMethodsSupported();
+        assertTrue(l.contains("S256"));
+        assertNull(emptyOidcProviderMetadata.getCodeChallengeMethodsSupported());
+    }
+
+    @Test
+    public void testTlsClientCertificateBoundAccessTokens() throws Exception {
+        assertTrue(oidcProviderMetadata.getTlsClientCertificateBoundAccessTokens());
+        assertFalse(withoutBooleansOidcProviderMetadata.getTlsClientCertificateBoundAccessTokens());
+    }
+
+    @Test
+    public void testRequestObjectEncryptionEncValuesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getRequestObjectEncryptionEncValuesSupported();
+        assertTrue(l.contains("A192GCM"));
+        assertNull(emptyOidcProviderMetadata.getRequestObjectEncryptionEncValuesSupported());
```

Comment:
I think this needs an assertNull for withoutBooleansProviderMetaData.

Reply:
done